signer/keydbstore/rethink_keydbstore.go

```diff
@@ -241,6 +266,45 @@ func (rdb RethinkDBKeyStore) RotateKeyPassphrase(keyID, newPassphraseAlias strin
 	return nil
 }
 
+// markActive marks a particular key as active
+func (rdb RethinkDBKeyStore) markActive(keyID string) error {
+	_, err := gorethink.DB(rdb.dbName).Table(PrivateKeysRethinkTable.Name).Get(keyID).Update(map[string]interface{}{
+		"last_used": rdb.nowFunc(),
+	}).RunWrite(rdb.sess)
+	return err
+}
+
+// Create will attempt to first re-use an inactive key for the same role, gun, and algorithm.
+// If one isn't found, it will create a private key and add it to the DB as an inactive key
+func (rdb RethinkDBKeyStore) Create(role, gun, algorithm string) (data.PublicKey, error) {
+	dbPrivateKey := RDBPrivateKey{}
+	res, err := gorethink.DB(rdb.dbName).Table(dbPrivateKey.TableName()).
+		Filter(gorethink.Row.Field("gun").Eq(gun)).
+		Filter(gorethink.Row.Field("role").Eq(role)).
+		Filter(gorethink.Row.Field("algorithm").Eq(algorithm)).
+		Filter(gorethink.Row.Field("last_used").Eq(time.Time{})).
+		Run(rdb.sess)
```

Comment:
group review: we might want to sort by rethink ID just so if there are multiple unused keys, calling create multiple times returns the same value rather than switching back enough.